src/Pact/Runtime/Capabilities.hs

```diff
@@ -51,19 +49,27 @@ type ApplyMgrFun e = Def Ref -> [PactValue] -> [PactValue] -> Eval e (Either Pac
 noopApplyMgrFun :: ApplyMgrFun e
 noopApplyMgrFun _ mgd _ = return $ Right mgd
 
-
+-- | Get any cap that is currently granted, of any scope.
 grantedCaps :: Eval e (S.Set Capability)
-grantedCaps = S.fromList . toList <$> use evalCapabilities
+grantedCaps = toSet <$> getAllStackCaps <*> getAllManaged
+  where
+    toSet a b = S.fromList $ a ++ b
+    getAllManaged = concatMap toList . S.toList <$> use (evalCapabilities . capManaged)
```

Comment:
`concatMap` is a member of `Foldable`, so `Set` will implement this. We can keep both `getAllStackCaps` and `getAllManaged` in `Set` and remove some O(n) behaviors.

Reply:
👍